wasm/mix-fetch/go-mix-conn/internal/jstypes/conv/request.go

```diff
@@ -223,23 +223,32 @@ func parseHeaders(headers js.Value, reqOpts types.RequestOptions, method string)
 func parseBody(request *js.Value) (io.Reader, error) {
 	jsBody := request.Get(fieldRequestBody)
 	var bodyReader io.Reader
+
 	if !jsBody.IsUndefined() && !jsBody.IsNull() {
-		log.Debug("stream body - getReader")
-		bodyReader = external.NewStreamReader(jsBody.Call("getReader"))
+		// Check to see if getReader is a function
+		if jsBody.InstanceOf(js.Global().Get("ReadableStream")) && jsBody.Get("getReader").Type() == js.TypeFunction {
+			log.Debug("stream body - getReader")
+			bodyReader = external.NewStreamReader(jsBody.Call("getReader"))
+		} else {
+			// Fall back to using ArrayBuffer
+			// https://developer.mozilla.org/en-US/docs/Web/API/Body/arrayBuffer
+			log.Debug("getReader not available - fallback to ArrayBuffer")
+			bodyReader = external.NewArrayReader(request.Call("arrayBuffer"))
+		}
 	} else {
-		log.Debug("unstremable body - fallback to ArrayBuffer")
-		// Fall back to using ArrayBuffer
-		// https://developer.mozilla.org/en-US/docs/Web/API/Body/arrayBuffer
+		log.Debug("unstreamable body - fallback to ArrayBuffer")
 		bodyReader = external.NewArrayReader(request.Call("arrayBuffer"))
 	}
```

Comment:
I **think** you might be able to collapse the nested if by removing the outer layer since, unless JS is doing something crazy, checking whether the body is an instance of `ReadableStream` should implicitly check if its null/undefined
```suggestion
	if jsBody.InstanceOf(js.Global().Get("ReadableStream")) && jsBody.Get("getReader").Type() == js.TypeFunction { 
		// Check to see if getReader is a function
		if jsBody.InstanceOf(js.Global().Get("ReadableStream")) && jsBody.Get("getReader").Type() == js.TypeFunction {
			log.Debug("stream body - getReader")
			bodyReader = external.NewStreamReader(jsBody.Call("getReader"))
	} else {
		log.Debug("unstreamable body - fallback to ArrayBuffer")
		bodyReader = external.NewArrayReader(request.Call("arrayBuffer"))
	}
```

Reply:
for some reason, github ui is not being nice here... amended to your suggestions after testing locally, all good.